pkg/apis/core/validation/validation.go

```diff
@@ -2286,6 +2289,41 @@ func validateDNSPolicy(dnsPolicy *core.DNSPolicy, fldPath *field.Path) field.Err
 	return allErrors
 }
 
+const (
+	maxDNSNameservers     = 3
+	maxDNSSearchPaths     = 6
+	maxDNSSearchListChars = 256
+)
+
+func validatePodDNSParams(dnsParams *core.PodDNSParams, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if dnsParams != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CustomPodDNS) {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "DNSParams: custom pod DNS is disabled by feature gate"))
+		} else {
+			if len(dnsParams.Nameservers) > maxDNSNameservers {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers"), dnsParams.Nameservers, fmt.Sprintf("must not have more than %v nameservers", maxDNSNameservers)))
+			}
+			for _, ns := range dnsParams.Nameservers {
+				if ip := net.ParseIP(ns); ip == nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers"), ns, "must be valid IP address"))
```

Comment:
You should track the index and pass that via fldPath.Child("nameservers").Index(i)

Same for all such loops

Reply:
Good catch, added index to fldpath.